// es-abstract/es2024.js
'use strict';

/* eslint global-require: 0 */
// https://262.ecma-international.org/15.0/#sec-abstract-operations
var ES2024 = {
	abs: require('es-abstract/2024/abst/2024/abs'),
	AddEntriesFromIterable: require('es-abstract/2024/AddEntriesFromIterableomIterable'),
	AddToKeptObjects: require('es-abstract/2024/AddToKeptObjectseptObjects'),
	AddValueToKeyedGroup: require('es-abstract/2024/AddValueToKeyedGroupKeyedGroup'),
	AdvanceStringIndex: require('es-abstract/2024/AdvanceStringIndextringIndex'),
	AllCharacters: require('es-abstract/2024/AllCharactersCharacters'),
	ApplyStringOrNumericBinaryOperator: require('es-abstract/2024/ApplyStringOrNumericBinaryOperatorryOperator'),
	ArrayBufferByteLength: require('es-abstract/2024/ArrayBufferByteLengthByteLength'),
	ArrayBufferCopyAndDetach: require('es-abstract/2024/ArrayBufferCopyAndDetachyAndDetach'),
	ArrayCreate: require('es-abstract/2024/ArrayCreaterrayCreate'),
	ArraySetLength: require('es-abstract/2024/ArraySetLengthySetLength'),
	ArraySpeciesCreate: require('es-abstract/2024/ArraySpeciesCreateciesCreate'),
	AsyncFromSyncIteratorContinuation: require('es-abstract/2024/AsyncFromSyncIteratorContinuationntinuation'),
	AsyncIteratorClose: require('es-abstract/2024/AsyncIteratorCloseratorClose'),
	BigInt: require('es-abstract/2024/BigInt024/BigInt'),
	BigIntBitwiseOp: require('es-abstract/2024/BigIntBitwiseOptBitwiseOp'),
	BinaryAnd: require('es-abstract/2024/BinaryAnd/BinaryAnd'),
	BinaryOr: require('es-abstract/2024/BinaryOr4/BinaryOr'),
	BinaryXor: require('es-abstract/2024/BinaryXor/BinaryXor'),
	ByteListBitwiseOp: require('es-abstract/2024/ByteListBitwiseOptBitwiseOp'),
	ByteListEqual: require('es-abstract/2024/ByteListEqualeListEqual'),
	Call: require('es-abstract/2024/Call/2024/Call'),
	CanBeHeldWeakly: require('es-abstract/2024/CanBeHeldWeaklyHeldWeakly'),
	Canonicalize: require('es-abstract/2024/Canonicalizenonicalize'),
	CanonicalNumericIndexString: require('es-abstract/2024/CanonicalNumericIndexStringndexString'),
	CharacterComplement: require('es-abstract/2024/CharacterComplementComplement'),
	CharacterRange: require('es-abstract/2024/CharacterRangeacterRange'),
	clamp: require('es-abstract/2024/clamp2024/clamp'),
	ClearKeptObjects: require('es-abstract/2024/ClearKeptObjectseptObjects'),
	CloneArrayBuffer: require('es-abstract/2024/CloneArrayBufferrrayBuffer'),
	CodePointAt: require('es-abstract/2024/CodePointAtodePointAt'),
	CodePointsToString: require('es-abstract/2024/CodePointsToStringtsToString'),
	CompareArrayElements: require('es-abstract/2024/CompareArrayElementsayElements'),
	CompareTypedArrayElements: require('es-abstract/2024/CompareTypedArrayElementsayElements'),
	CompletePropertyDescriptor: require('es-abstract/2024/CompletePropertyDescriptorDescriptor'),
	CompletionRecord: require('es-abstract/2024/CompletionRecordtionRecord'),
	CopyDataProperties: require('es-abstract/2024/CopyDataPropertiesProperties'),
	CreateAsyncFromSyncIterator: require('es-abstract/2024/CreateAsyncFromSyncIteratorncIterator'),
	CreateDataProperty: require('es-abstract/2024/CreateDataPropertytaProperty'),
	CreateDataPropertyOrThrow: require('es-abstract/2024/CreateDataPropertyOrThrowrtyOrThrow'),
	CreateHTML: require('es-abstract/2024/CreateHTMLCreateHTML'),
	CreateIterResultObject: require('es-abstract/2024/CreateIterResultObjectsultObject'),
	CreateListFromArrayLike: require('es-abstract/2024/CreateListFromArrayLikemArrayLike'),
	CreateNonEnumerableDataPropertyOrThrow: require('es-abstract/2024/CreateNonEnumerableDataPropertyOrThrowrtyOrThrow'),
	CreateRegExpStringIterator: require('es-abstract/2024/CreateRegExpStringIteratorngIterator'),
	DateFromTime: require('es-abstract/2024/DateFromTimeteFromTime'),
	DateString: require('es-abstract/2024/DateStringDateString'),
	Day: require('es-abstract/2024/Dayt/2024/Day'),
	DayFromYear: require('es-abstract/2024/DayFromYearayFromYear'),
	DaysInYear: require('es-abstract/2024/DaysInYearDaysInYear'),
	DayWithinYear: require('es-abstract/2024/DayWithinYearWithinYear'),
	DefineMethodProperty: require('es-abstract/2024/DefineMethodPropertyodProperty'),
	DefinePropertyOrThrow: require('es-abstract/2024/DefinePropertyOrThrowrtyOrThrow'),
	DeletePropertyOrThrow: require('es-abstract/2024/DeletePropertyOrThrowrtyOrThrow'),
	DetachArrayBuffer: require('es-abstract/2024/DetachArrayBufferrrayBuffer'),
	EnumerableOwnProperties: require('es-abstract/2024/EnumerableOwnPropertiesProperties'),
	FindViaPredicate: require('es-abstract/2024/FindViaPredicateaPredicate'),
	FlattenIntoArray: require('es-abstract/2024/FlattenIntoArraynIntoArray'),
	floor: require('es-abstract/2024/floor2024/floor'),
	FromPropertyDescriptor: require('es-abstract/2024/FromPropertyDescriptorDescriptor'),
	Get: require('es-abstract/2024/Gett/2024/Get'),
	GetArrayBufferMaxByteLengthOption: require('es-abstract/2024/GetArrayBufferMaxByteLengthOptionngthOption'),
	GetGlobalObject: require('es-abstract/2024/GetGlobalObjectobalObject'),
	GetIterator: require('es-abstract/2024/GetIteratoretIterator'),
	GetIteratorFromMethod: require('es-abstract/2024/GetIteratorFromMethodFromMethod'),
	GetMatchIndexPair: require('es-abstract/2024/GetMatchIndexPairhIndexPair'),
	GetMatchString: require('es-abstract/2024/GetMatchStringatchString'),
	GetMethod: require('es-abstract/2024/GetMethod/GetMethod'),
	GetNamedTimeZoneEpochNanoseconds: require('es-abstract/2024/GetNamedTimeZoneEpochNanosecondsanoseconds'),
	GetOwnPropertyKeys: require('es-abstract/2024/GetOwnPropertyKeysopertyKeys'),
	GetPromiseResolve: require('es-abstract/2024/GetPromiseResolveiseResolve'),
	GetPrototypeFromConstructor: require('es-abstract/2024/GetPrototypeFromConstructoronstructor'),
	GetStringIndex: require('es-abstract/2024/GetStringIndextringIndex'),
	GetSubstitution: require('es-abstract/2024/GetSubstitutionbstitution'),
	GetUTCEpochNanoseconds: require('es-abstract/2024/GetUTCEpochNanosecondsanoseconds'),
	GetV: require('es-abstract/2024/GetV/2024/GetV'),
	GetValueFromBuffer: require('es-abstract/2024/GetValueFromBufferFromBuffer'),
	GetViewByteLength: require('es-abstract/2024/GetViewByteLengthByteLength'),
	GroupBy: require('es-abstract/2024/GroupBy24/GroupBy'),
	HasEitherUnicodeFlag: require('es-abstract/2024/HasEitherUnicodeFlagnicodeFlag'),
	HasOwnProperty: require('es-abstract/2024/HasOwnPropertywnProperty'),
	HasProperty: require('es-abstract/2024/HasPropertyasProperty'),
	HourFromTime: require('es-abstract/2024/HourFromTimeurFromTime'),
	InLeapYear: require('es-abstract/2024/InLeapYearInLeapYear'),
	InstallErrorCause: require('es-abstract/2024/InstallErrorCauseErrorCause'),
	InstanceofOperator: require('es-abstract/2024/InstanceofOperatorofOperator'),
	InternalizeJSONProperty: require('es-abstract/2024/InternalizeJSONPropertyONProperty'),
	Invoke: require('es-abstract/2024/Invoke024/Invoke'),
	IsAccessorDescriptor: require('es-abstract/2024/IsAccessorDescriptorDescriptor'),
	IsArray: require('es-abstract/2024/IsArray24/IsArray'),
	IsArrayBufferViewOutOfBounds: require('es-abstract/2024/IsArrayBufferViewOutOfBoundsutOfBounds'),
	IsBigIntElementType: require('es-abstract/2024/IsBigIntElementTypelementType'),
	IsCallable: require('es-abstract/2024/IsCallableIsCallable'),
	IsCompatiblePropertyDescriptor: require('es-abstract/2024/IsCompatiblePropertyDescriptorDescriptor'),
	IsConcatSpreadable: require('es-abstract/2024/IsConcatSpreadableSpreadable'),
	IsConstructor: require('es-abstract/2024/IsConstructoronstructor'),
	IsDataDescriptor: require('es-abstract/2024/IsDataDescriptorDescriptor'),
	IsDetachedBuffer: require('es-abstract/2024/IsDetachedBufferchedBuffer'),
	IsExtensible: require('es-abstract/2024/IsExtensibleExtensible'),
	IsFixedLengthArrayBuffer: require('es-abstract/2024/IsFixedLengthArrayBufferrrayBuffer'),
	IsGenericDescriptor: require('es-abstract/2024/IsGenericDescriptorDescriptor'),
	IsIntegralNumber: require('es-abstract/2024/IsIntegralNumbergralNumber'),
	IsLessThan: require('es-abstract/2024/IsLessThanIsLessThan'),
	IsLooselyEqual: require('es-abstract/2024/IsLooselyEqualoselyEqual'),
	IsNoTearConfiguration: require('es-abstract/2024/IsNoTearConfigurationfiguration'),
	IsPromise: require('es-abstract/2024/IsPromise/IsPromise'),
	IsPropertyKey: require('es-abstract/2024/IsPropertyKeyropertyKey'),
	IsRegExp: require('es-abstract/2024/IsRegExp4/IsRegExp'),
	IsSharedArrayBuffer: require('es-abstract/2024/IsSharedArrayBufferrrayBuffer'),
	IsStrictlyEqual: require('es-abstract/2024/IsStrictlyEqualictlyEqual'),
	IsStringWellFormedUnicode: require('es-abstract/2024/IsStringWellFormedUnicodemedUnicode'),
	IsTimeZoneOffsetString: require('es-abstract/2024/IsTimeZoneOffsetStringfsetString'),
	IsTypedArrayOutOfBounds: require('es-abstract/2024/IsTypedArrayOutOfBoundsutOfBounds'),
	IsUnclampedIntegerElementType: require('es-abstract/2024/IsUnclampedIntegerElementTypelementType'),
	IsUnsignedElementType: require('es-abstract/2024/IsUnsignedElementTypelementType'),
	IsValidIntegerIndex: require('es-abstract/2024/IsValidIntegerIndextegerIndex'),
	IsViewOutOfBounds: require('es-abstract/2024/IsViewOutOfBoundsutOfBounds'),
	IsWordChar: require('es-abstract/2024/IsWordCharIsWordChar'),
	IteratorClose: require('es-abstract/2024/IteratorCloseratorClose'),
	IteratorComplete: require('es-abstract/2024/IteratorCompleteorComplete'),
	IteratorNext: require('es-abstract/2024/IteratorNexteratorNext'),
	IteratorStep: require('es-abstract/2024/IteratorSteperatorStep'),
	IteratorStepValue: require('es-abstract/2024/IteratorStepValuerStepValue'),
	IteratorToList: require('es-abstract/2024/IteratorToListatorToList'),
	IteratorValue: require('es-abstract/2024/IteratorValueratorValue'),
	KeyForSymbol: require('es-abstract/2024/KeyForSymbolyForSymbol'),
	LengthOfArrayLike: require('es-abstract/2024/LengthOfArrayLikefArrayLike'),
	MakeDataViewWithBufferWitnessRecord: require('es-abstract/2024/MakeDataViewWithBufferWitnessRecordnessRecord'),
	MakeDate: require('es-abstract/2024/MakeDate4/MakeDate'),
	MakeDay: require('es-abstract/2024/MakeDay24/MakeDay'),
	MakeFullYear: require('es-abstract/2024/MakeFullYearkeFullYear'),
	MakeMatchIndicesIndexPairArray: require('es-abstract/2024/MakeMatchIndicesIndexPairArrayxPairArray'),
	MakeTime: require('es-abstract/2024/MakeTime4/MakeTime'),
	MakeTypedArrayWithBufferWitnessRecord: require('es-abstract/2024/MakeTypedArrayWithBufferWitnessRecordnessRecord'),
	max: require('es-abstract/2024/maxt/2024/max'),
	min: require('es-abstract/2024/mint/2024/min'),
	MinFromTime: require('es-abstract/2024/MinFromTimeinFromTime'),
	modulo: require('es-abstract/2024/modulo024/modulo'),
	MonthFromTime: require('es-abstract/2024/MonthFromTimethFromTime'),
	msFromTime: require('es-abstract/2024/msFromTimemsFromTime'),
	NewPromiseCapability: require('es-abstract/2024/NewPromiseCapabilityCapability'),
	NormalCompletion: require('es-abstract/2024/NormalCompletionCompletion'),
	Number: require('es-abstract/2024/Number024/Number'),
	NumberBitwiseOp: require('es-abstract/2024/NumberBitwiseOprBitwiseOp'),
	NumberToBigInt: require('es-abstract/2024/NumberToBigInterToBigInt'),
	NumericToRawBytes: require('es-abstract/2024/NumericToRawBytesToRawBytes'),
	ObjectDefineProperties: require('es-abstract/2024/ObjectDefinePropertiesProperties'),
	OrdinaryCreateFromConstructor: require('es-abstract/2024/OrdinaryCreateFromConstructoronstructor'),
	OrdinaryDefineOwnProperty: require('es-abstract/2024/OrdinaryDefineOwnPropertywnProperty'),
	OrdinaryGetOwnProperty: require('es-abstract/2024/OrdinaryGetOwnPropertywnProperty'),
	OrdinaryGetPrototypeOf: require('es-abstract/2024/OrdinaryGetPrototypeOfrototypeOf'),
	OrdinaryHasInstance: require('es-abstract/2024/OrdinaryHasInstanceasInstance'),
	OrdinaryHasProperty: require('es-abstract/2024/OrdinaryHasPropertyasProperty'),
	OrdinaryObjectCreate: require('es-abstract/2024/OrdinaryObjectCreatejectCreate'),
	OrdinarySetPrototypeOf: require('es-abstract/2024/OrdinarySetPrototypeOfrototypeOf'),
	OrdinaryToPrimitive: require('es-abstract/2024/OrdinaryToPrimitiveoPrimitive'),
	ParseHexOctet: require('es-abstract/2024/ParseHexOctetseHexOctet'),
	PromiseResolve: require('es-abstract/2024/PromiseResolveiseResolve'),
	QuoteJSONString: require('es-abstract/2024/QuoteJSONStringJSONString'),
	RawBytesToNumeric: require('es-abstract/2024/RawBytesToNumericsToNumeric'),
	RegExpCreate: require('es-abstract/2024/RegExpCreategExpCreate'),
	RegExpExec: require('es-abstract/2024/RegExpExecRegExpExec'),
	RegExpHasFlag: require('es-abstract/2024/RegExpHasFlagExpHasFlag'),
	RequireObjectCoercible: require('es-abstract/2024/RequireObjectCoercibletCoercible'),
	SameValue: require('es-abstract/2024/SameValue/SameValue'),
	SameValueNonNumber: require('es-abstract/2024/SameValueNonNumbereNonNumber'),
	SameValueZero: require('es-abstract/2024/SameValueZeroeValueZero'),
	SecFromTime: require('es-abstract/2024/SecFromTimeecFromTime'),
	Set: require('es-abstract/2024/Sett/2024/Set'),
	SetFunctionLength: require('es-abstract/2024/SetFunctionLengthtionLength'),
	SetFunctionName: require('es-abstract/2024/SetFunctionNamenctionName'),
	SetIntegrityLevel: require('es-abstract/2024/SetIntegrityLevelgrityLevel'),
	SetTypedArrayFromArrayLike: require('es-abstract/2024/SetTypedArrayFromArrayLikemArrayLike'),
	SetTypedArrayFromTypedArray: require('es-abstract/2024/SetTypedArrayFromTypedArrayTypedArray'),
	SetValueInBuffer: require('es-abstract/2024/SetValueInBufferueInBuffer'),
	SortIndexedProperties: require('es-abstract/2024/SortIndexedPropertiesProperties'),
	SpeciesConstructor: require('es-abstract/2024/SpeciesConstructoronstructor'),
	StringCreate: require('es-abstract/2024/StringCreateringCreate'),
	StringGetOwnProperty: require('es-abstract/2024/StringGetOwnPropertywnProperty'),
	StringIndexOf: require('es-abstract/2024/StringIndexOfingIndexOf'),
	StringPad: require('es-abstract/2024/StringPad/StringPad'),
	StringPaddingBuiltinsImpl: require('es-abstract/2024/StringPaddingBuiltinsImpliltinsImpl'),
	StringToBigInt: require('es-abstract/2024/StringToBigIntngToBigInt'),
	StringToCodePoints: require('es-abstract/2024/StringToCodePointsCodePoints'),
	StringToNumber: require('es-abstract/2024/StringToNumberngToNumber'),
	substring: require('es-abstract/2024/substring/substring'),
	SymbolDescriptiveString: require('es-abstract/2024/SymbolDescriptiveStringtiveString'),
	SystemTimeZoneIdentifier: require('es-abstract/2024/SystemTimeZoneIdentifierIdentifier'),
	TestIntegrityLevel: require('es-abstract/2024/TestIntegrityLevelgrityLevel'),
	ThisBigIntValue: require('es-abstract/2024/ThisBigIntValueigIntValue'),
	ThisBooleanValue: require('es-abstract/2024/ThisBooleanValueoleanValue'),
	ThisNumberValue: require('es-abstract/2024/ThisNumberValueumberValue'),
	ThisStringValue: require('es-abstract/2024/ThisStringValuetringValue'),
	ThisSymbolValue: require('es-abstract/2024/ThisSymbolValueymbolValue'),
	ThrowCompletion: require('es-abstract/2024/ThrowCompletionCompletion'),
	TimeClip: require('es-abstract/2024/TimeClip4/TimeClip'),
	TimeFromYear: require('es-abstract/2024/TimeFromYearmeFromYear'),
	TimeString: require('es-abstract/2024/TimeStringTimeString'),
	TimeWithinDay: require('es-abstract/2024/TimeWithinDayeWithinDay'),
	TimeZoneString: require('es-abstract/2024/TimeZoneStringZoneString'),
	ToBigInt: require('es-abstract/2024/ToBigInt4/ToBigInt'),
	ToBigInt64: require('es-abstract/2024/ToBigInt64ToBigInt64'),
	ToBigUint64: require('es-abstract/2024/ToBigUint64oBigUint64'),
	ToBoolean: require('es-abstract/2024/ToBoolean/ToBoolean'),
	ToDateString: require('es-abstract/2024/ToDateStringDateString'),
	ToIndex: require('es-abstract/2024/ToIndex24/ToIndex'),
	ToInt16: require('es-abstract/2024/ToInt1624/ToInt16'),
	ToInt32: require('es-abstract/2024/ToInt3224/ToInt32'),
	ToInt8: require('es-abstract/2024/ToInt8024/ToInt8'),
	ToIntegerOrInfinity: require('es-abstract/2024/ToIntegerOrInfinityOrInfinity'),
	ToLength: require('es-abstract/2024/ToLength4/ToLength'),
	ToNumber: require('es-abstract/2024/ToNumber4/ToNumber'),
	ToNumeric: require('es-abstract/2024/ToNumeric/ToNumeric'),
	ToObject: require('es-abstract/2024/ToObject4/ToObject'),
	ToPrimitive: require('es-abstract/2024/ToPrimitiveoPrimitive'),
	ToPropertyDescriptor: require('es-abstract/2024/ToPropertyDescriptorDescriptor'),
	ToPropertyKey: require('es-abstract/2024/ToPropertyKeyropertyKey'),
	ToString: require('es-abstract/2024/ToString4/ToString'),
	ToUint16: require('es-abstract/2024/ToUint164/ToUint16'),
	ToUint32: require('es-abstract/2024/ToUint324/ToUint32'),
	ToUint8: require('es-abstract/2024/ToUint824/ToUint8'),
	ToUint8Clamp: require('es-abstract/2024/ToUint8ClampUint8Clamp'),
	ToZeroPaddedDecimalString: require('es-abstract/2024/ToZeroPaddedDecimalStringimalString'),
	TrimString: require('es-abstract/2024/TrimStringTrimString'),
	truncate: require('es-abstract/2024/truncate4/truncate'),
	Type: require('es-abstract/2024/Type/2024/Type'),
	TypedArrayByteLength: require('es-abstract/2024/TypedArrayByteLengthByteLength'),
	TypedArrayCreateFromConstructor: require('es-abstract/2024/TypedArrayCreateFromConstructoronstructor'),
	TypedArrayCreateSameType: require('es-abstract/2024/TypedArrayCreateSameTypeteSameType'),
	TypedArrayElementSize: require('es-abstract/2024/TypedArrayElementSizelementSize'),
	TypedArrayElementType: require('es-abstract/2024/TypedArrayElementTypelementType'),
	TypedArrayGetElement: require('es-abstract/2024/TypedArrayGetElementGetElement'),
	TypedArrayLength: require('es-abstract/2024/TypedArrayLengthrrayLength'),
	TypedArraySetElement: require('es-abstract/2024/TypedArraySetElementSetElement'),
	TypedArraySpeciesCreate: require('es-abstract/2024/TypedArraySpeciesCreateciesCreate'),
	UnicodeEscape: require('es-abstract/2024/UnicodeEscapecodeEscape'),
	UTF16EncodeCodePoint: require('es-abstract/2024/UTF16EncodeCodePointeCodePoint'),
	UTF16SurrogatePairToCodePoint: require('es-abstract/2024/UTF16SurrogatePairToCodePointoCodePoint'),
	ValidateAndApplyPropertyDescriptor: require('es-abstract/2024/ValidateAndApplyPropertyDescriptorDescriptor'),
	ValidateAtomicAccess: require('es-abstract/2024/ValidateAtomicAccessomicAccess'),
	ValidateAtomicAccessOnIntegerTypedArray: require('es-abstract/2024/ValidateAtomicAccessOnIntegerTypedArrayTypedArray'),
	ValidateIntegerTypedArray: require('es-abstract/2024/ValidateIntegerTypedArrayTypedArray'),
	ValidateTypedArray: require('es-abstract/2024/ValidateTypedArrayTypedArray'),
	WeakRefDeref: require('es-abstract/2024/WeakRefDerefakRefDeref'),
	WeekDay: require('es-abstract/2024/WeekDay24/WeekDay'),
	WordCharacters: require('es-abstract/2024/WordCharactersCharacters'),
	YearFromTime: require('es-abstract/2024/YearFromTimearFromTime')
};

module.exports = ES2024;
