// es-abstract/es2017.js
'use strict';

/* eslint global-require: 0 */
// https://262.ecma-international.org/8.0/#sec-abstract-operations
var ES2017 = {
	'Abstract Equality Comparison': require('es-abstract/2017/AbstractEqualityComparisonComparison'),
	'Abstract Relational Comparison': require('es-abstract/2017/AbstractRelationalComparisonComparison'),
	'Strict Equality Comparison': require('es-abstract/2017/StrictEqualityComparisonComparison'),
	abs: require('es-abstract/2017/abst/2017/abs'),
	AdvanceStringIndex: require('es-abstract/2017/AdvanceStringIndextringIndex'),
	ArrayCreate: require('es-abstract/2017/ArrayCreaterrayCreate'),
	ArraySetLength: require('es-abstract/2017/ArraySetLengthySetLength'),
	ArraySpeciesCreate: require('es-abstract/2017/ArraySpeciesCreateciesCreate'),
	Call: require('es-abstract/2017/Call/2017/Call'),
	Canonicalize: require('es-abstract/2017/Canonicalizenonicalize'),
	CanonicalNumericIndexString: require('es-abstract/2017/CanonicalNumericIndexStringndexString'),
	CharacterRange: require('es-abstract/2017/CharacterRangeacterRange'),
	CompletePropertyDescriptor: require('es-abstract/2017/CompletePropertyDescriptorDescriptor'),
	CompletionRecord: require('es-abstract/2017/CompletionRecordtionRecord'),
	CreateDataProperty: require('es-abstract/2017/CreateDataPropertytaProperty'),
	CreateDataPropertyOrThrow: require('es-abstract/2017/CreateDataPropertyOrThrowrtyOrThrow'),
	CreateHTML: require('es-abstract/2017/CreateHTMLCreateHTML'),
	CreateIterResultObject: require('es-abstract/2017/CreateIterResultObjectsultObject'),
	CreateListFromArrayLike: require('es-abstract/2017/CreateListFromArrayLikemArrayLike'),
	CreateMethodProperty: require('es-abstract/2017/CreateMethodPropertyodProperty'),
	DateFromTime: require('es-abstract/2017/DateFromTimeteFromTime'),
	Day: require('es-abstract/2017/Dayt/2017/Day'),
	DayFromYear: require('es-abstract/2017/DayFromYearayFromYear'),
	DaysInYear: require('es-abstract/2017/DaysInYearDaysInYear'),
	DayWithinYear: require('es-abstract/2017/DayWithinYearWithinYear'),
	DefinePropertyOrThrow: require('es-abstract/2017/DefinePropertyOrThrowrtyOrThrow'),
	DeletePropertyOrThrow: require('es-abstract/2017/DeletePropertyOrThrowrtyOrThrow'),
	DetachArrayBuffer: require('es-abstract/2017/DetachArrayBufferrrayBuffer'),
	EnumerableOwnProperties: require('es-abstract/2017/EnumerableOwnPropertiesProperties'),
	floor: require('es-abstract/2017/floor2017/floor'),
	FromPropertyDescriptor: require('es-abstract/2017/FromPropertyDescriptorDescriptor'),
	Get: require('es-abstract/2017/Gett/2017/Get'),
	GetGlobalObject: require('es-abstract/2017/GetGlobalObjectobalObject'),
	GetIterator: require('es-abstract/2017/GetIteratoretIterator'),
	GetMethod: require('es-abstract/2017/GetMethod/GetMethod'),
	GetOwnPropertyKeys: require('es-abstract/2017/GetOwnPropertyKeysopertyKeys'),
	GetPrototypeFromConstructor: require('es-abstract/2017/GetPrototypeFromConstructoronstructor'),
	GetSubstitution: require('es-abstract/2017/GetSubstitutionbstitution'),
	GetV: require('es-abstract/2017/GetV/2017/GetV'),
	GetValueFromBuffer: require('es-abstract/2017/GetValueFromBufferFromBuffer'),
	HasOwnProperty: require('es-abstract/2017/HasOwnPropertywnProperty'),
	HasProperty: require('es-abstract/2017/HasPropertyasProperty'),
	HourFromTime: require('es-abstract/2017/HourFromTimeurFromTime'),
	InLeapYear: require('es-abstract/2017/InLeapYearInLeapYear'),
	InstanceofOperator: require('es-abstract/2017/InstanceofOperatorofOperator'),
	IntegerIndexedElementGet: require('es-abstract/2017/IntegerIndexedElementGetElementGet'),
	IntegerIndexedElementSet: require('es-abstract/2017/IntegerIndexedElementSetElementSet'),
	InternalizeJSONProperty: require('es-abstract/2017/InternalizeJSONPropertyONProperty'),
	Invoke: require('es-abstract/2017/Invoke017/Invoke'),
	IsAccessorDescriptor: require('es-abstract/2017/IsAccessorDescriptorDescriptor'),
	IsArray: require('es-abstract/2017/IsArray17/IsArray'),
	IsCallable: require('es-abstract/2017/IsCallableIsCallable'),
	IsCompatiblePropertyDescriptor: require('es-abstract/2017/IsCompatiblePropertyDescriptorDescriptor'),
	IsConcatSpreadable: require('es-abstract/2017/IsConcatSpreadableSpreadable'),
	IsConstructor: require('es-abstract/2017/IsConstructoronstructor'),
	IsDataDescriptor: require('es-abstract/2017/IsDataDescriptorDescriptor'),
	IsDetachedBuffer: require('es-abstract/2017/IsDetachedBufferchedBuffer'),
	IsExtensible: require('es-abstract/2017/IsExtensibleExtensible'),
	IsGenericDescriptor: require('es-abstract/2017/IsGenericDescriptorDescriptor'),
	IsInteger: require('es-abstract/2017/IsInteger/IsInteger'),
	IsPromise: require('es-abstract/2017/IsPromise/IsPromise'),
	IsPropertyDescriptor: require('es-abstract/2017/IsPropertyDescriptorDescriptor'),
	IsPropertyKey: require('es-abstract/2017/IsPropertyKeyropertyKey'),
	IsRegExp: require('es-abstract/2017/IsRegExp7/IsRegExp'),
	IsSharedArrayBuffer: require('es-abstract/2017/IsSharedArrayBufferrrayBuffer'),
	IsWordChar: require('es-abstract/2017/IsWordCharIsWordChar'),
	IterableToList: require('es-abstract/2017/IterableToListableToList'),
	IteratorClose: require('es-abstract/2017/IteratorCloseratorClose'),
	IteratorComplete: require('es-abstract/2017/IteratorCompleteorComplete'),
	IteratorNext: require('es-abstract/2017/IteratorNexteratorNext'),
	IteratorStep: require('es-abstract/2017/IteratorSteperatorStep'),
	IteratorValue: require('es-abstract/2017/IteratorValueratorValue'),
	MakeDate: require('es-abstract/2017/MakeDate7/MakeDate'),
	MakeDay: require('es-abstract/2017/MakeDay17/MakeDay'),
	MakeTime: require('es-abstract/2017/MakeTime7/MakeTime'),
	max: require('es-abstract/2017/maxt/2017/max'),
	min: require('es-abstract/2017/mint/2017/min'),
	MinFromTime: require('es-abstract/2017/MinFromTimeinFromTime'),
	modulo: require('es-abstract/2017/modulo017/modulo'),
	MonthFromTime: require('es-abstract/2017/MonthFromTimethFromTime'),
	msFromTime: require('es-abstract/2017/msFromTimemsFromTime'),
	NewPromiseCapability: require('es-abstract/2017/NewPromiseCapabilityCapability'),
	NormalCompletion: require('es-abstract/2017/NormalCompletionCompletion'),
	NumberToRawBytes: require('es-abstract/2017/NumberToRawBytesToRawBytes'),
	ObjectCreate: require('es-abstract/2017/ObjectCreatejectCreate'),
	ObjectDefineProperties: require('es-abstract/2017/ObjectDefinePropertiesProperties'),
	OrdinaryCreateFromConstructor: require('es-abstract/2017/OrdinaryCreateFromConstructoronstructor'),
	OrdinaryDefineOwnProperty: require('es-abstract/2017/OrdinaryDefineOwnPropertywnProperty'),
	OrdinaryGetOwnProperty: require('es-abstract/2017/OrdinaryGetOwnPropertywnProperty'),
	OrdinaryGetPrototypeOf: require('es-abstract/2017/OrdinaryGetPrototypeOfrototypeOf'),
	OrdinaryHasInstance: require('es-abstract/2017/OrdinaryHasInstanceasInstance'),
	OrdinaryHasProperty: require('es-abstract/2017/OrdinaryHasPropertyasProperty'),
	OrdinarySetPrototypeOf: require('es-abstract/2017/OrdinarySetPrototypeOfrototypeOf'),
	OrdinaryToPrimitive: require('es-abstract/2017/OrdinaryToPrimitiveoPrimitive'),
	QuoteJSONString: require('es-abstract/2017/QuoteJSONStringJSONString'),
	RawBytesToNumber: require('es-abstract/2017/RawBytesToNumberesToNumber'),
	RegExpCreate: require('es-abstract/2017/RegExpCreategExpCreate'),
	RegExpExec: require('es-abstract/2017/RegExpExecRegExpExec'),
	RequireObjectCoercible: require('es-abstract/2017/RequireObjectCoercibletCoercible'),
	SameValue: require('es-abstract/2017/SameValue/SameValue'),
	SameValueNonNumber: require('es-abstract/2017/SameValueNonNumbereNonNumber'),
	SameValueZero: require('es-abstract/2017/SameValueZeroeValueZero'),
	SecFromTime: require('es-abstract/2017/SecFromTimeecFromTime'),
	Set: require('es-abstract/2017/Sett/2017/Set'),
	SetFunctionName: require('es-abstract/2017/SetFunctionNamenctionName'),
	SetIntegrityLevel: require('es-abstract/2017/SetIntegrityLevelgrityLevel'),
	SetValueInBuffer: require('es-abstract/2017/SetValueInBufferueInBuffer'),
	SpeciesConstructor: require('es-abstract/2017/SpeciesConstructoronstructor'),
	SplitMatch: require('es-abstract/2017/SplitMatchSplitMatch'),
	StringCreate: require('es-abstract/2017/StringCreateringCreate'),
	StringGetOwnProperty: require('es-abstract/2017/StringGetOwnPropertywnProperty'),
	SymbolDescriptiveString: require('es-abstract/2017/SymbolDescriptiveStringtiveString'),
	TestIntegrityLevel: require('es-abstract/2017/TestIntegrityLevelgrityLevel'),
	thisBooleanValue: require('es-abstract/2017/thisBooleanValueoleanValue'),
	thisNumberValue: require('es-abstract/2017/thisNumberValueumberValue'),
	thisStringValue: require('es-abstract/2017/thisStringValuetringValue'),
	thisTimeValue: require('es-abstract/2017/thisTimeValuesTimeValue'),
	TimeClip: require('es-abstract/2017/TimeClip7/TimeClip'),
	TimeFromYear: require('es-abstract/2017/TimeFromYearmeFromYear'),
	TimeWithinDay: require('es-abstract/2017/TimeWithinDayeWithinDay'),
	ToBoolean: require('es-abstract/2017/ToBoolean/ToBoolean'),
	ToDateString: require('es-abstract/2017/ToDateStringDateString'),
	ToIndex: require('es-abstract/2017/ToIndex17/ToIndex'),
	ToInt16: require('es-abstract/2017/ToInt1617/ToInt16'),
	ToInt32: require('es-abstract/2017/ToInt3217/ToInt32'),
	ToInt8: require('es-abstract/2017/ToInt8017/ToInt8'),
	ToInteger: require('es-abstract/2017/ToInteger/ToInteger'),
	ToLength: require('es-abstract/2017/ToLength7/ToLength'),
	ToNumber: require('es-abstract/2017/ToNumber7/ToNumber'),
	ToObject: require('es-abstract/2017/ToObject7/ToObject'),
	ToPrimitive: require('es-abstract/2017/ToPrimitiveoPrimitive'),
	ToPropertyDescriptor: require('es-abstract/2017/ToPropertyDescriptorDescriptor'),
	ToPropertyKey: require('es-abstract/2017/ToPropertyKeyropertyKey'),
	ToString: require('es-abstract/2017/ToString7/ToString'),
	ToUint16: require('es-abstract/2017/ToUint167/ToUint16'),
	ToUint32: require('es-abstract/2017/ToUint327/ToUint32'),
	ToUint8: require('es-abstract/2017/ToUint817/ToUint8'),
	ToUint8Clamp: require('es-abstract/2017/ToUint8ClampUint8Clamp'),
	Type: require('es-abstract/2017/Type/2017/Type'),
	TypedArrayCreate: require('es-abstract/2017/TypedArrayCreaterrayCreate'),
	TypedArraySpeciesCreate: require('es-abstract/2017/TypedArraySpeciesCreateciesCreate'),
	UTF16Decode: require('es-abstract/2017/UTF16DecodeTF16Decode'),
	UTF16Encoding: require('es-abstract/2017/UTF16Encoding16Encoding'),
	ValidateAndApplyPropertyDescriptor: require('es-abstract/2017/ValidateAndApplyPropertyDescriptorDescriptor'),
	ValidateAtomicAccess: require('es-abstract/2017/ValidateAtomicAccessomicAccess'),
	ValidateTypedArray: require('es-abstract/2017/ValidateTypedArrayTypedArray'),
	WeekDay: require('es-abstract/2017/WeekDay17/WeekDay'),
	WordCharacters: require('es-abstract/2017/WordCharactersCharacters'),
	YearFromTime: require('es-abstract/2017/YearFromTimearFromTime')
};

module.exports = ES2017;
